File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/user/WithdrawalUserCreator.java
package com.depromeet.threedollar.domain.rds.user.domain.user;

import org.jetbrains.annotations.NotNull;

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
@TestFixture
public class WithdrawalUserCreator {

    @Builder
    public static WithdrawalUser create(
        @NotNull String socialId,
        @NotNull UserSocialType socialType,
        @NotNull Long userId,
        @NotNull String name
    ) {
        return WithdrawalUser.builder()
            .userId(userId)
            .socialInfo(SocialInfo.of(socialId, socialType))
            .name(name)
            .build();
    }

}

File: threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/user/WithdrawalUserCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.user

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture

@TestFixture
object WithdrawalUserCreator {

    @JvmOverloads
    @JvmStatic
    fun create(
            socialId: String,
            socialType: UserSocialType,
            userId: Long,
            name: String = "가삼",
    ): WithdrawalUser {
        return WithdrawalUser.builder()
            .userId(userId)
            .socialInfo(SocialInfo.of(socialId, socialType))
            .name(name)
            .build()
    }

}
